checkpoint_connector.py

```diff
@@ -201,46 +209,38 @@ def _login(self, action_result):
         if not ret_val:
             return action_result.get_status()
 
-        self._sid = resp_json.get('sid')
+        self._state['sid'] = resp_json.get('sid')
 
-        self._headers['X-chkp-sid'] = self._sid
+        self._headers['X-chkp-sid'] = self._state.get('sid')
+
+        self.save_state(self._state)
 
         return phantom.APP_SUCCESS
 
     def _logout(self, action_result):
 
-        if self._sid is None:
-            # logout already called, sid is null
-            return phantom.APP_SUCCESS
-
         ret_val, resp_json = self._make_rest_call('logout', {}, action_result)
 
         if phantom.is_fail(ret_val):
             self.save_progress("Failed to logout: {}".format(action_result.get_status_message()))
             return action_result.get_status(), action_result.get_status_message()
 
-        self._sid = None
-
         return phantom.APP_SUCCESS, "Successfully logged out of session"
 
     def _logout_session(self, param):
 
         action_result = self.add_action_result(ActionResult(dict(param)))
 
-        if not(self._login(action_result)):
+        if not(self._set_auth_sid(action_result)):
```

Comment:
```suggestion
        if not self._set_auth_sid(action_result):
```

ditto for the other `not()` cases later on